tests/functional/g/globals.py

```diff
@@ -102,3 +102,12 @@ def init_connection_state(alias):
     global RAN_DB_DICT  # [global-variable-not-assigned]
     RAN_DB_SET.add(alias)
     return RAN_DB_DICT.setdefault("color", "Palomino")
+
+
+# Prevent emitting `invalid-name` for the line on which `global` is declared
+# https://github.com/PyCQA/pylint/issues/8307
+
+_foo: str = "tomato"
```

Comment:
But it should be raised here, right ?
```suggestion
_foo: str = "tomato"  # [disallowed-name]
```

Reply:
Yeah sorry I'm confused about it, there's a lot of discussion around invalid-name, hard to know what's done or not (https://github.com/PyCQA/pylint/issues/7305). There's also almost too much to digest in https://pylint.readthedocs.io/en/latest/user_guide/messages/convention/invalid-name.html and yet no explanation about Final.

Reply:
We have a [global-variable-undefined](https://github.com/PyCQA/pylint/blob/a0b28f9019fabddd8ef428be75659082377abb4c/pylint/checkers/variables.py#L440) message which will be emitted when the global name is not defined in the module scope. So, when it is defined in the module scope the `invalid-name` rule-checks are triggered by `visit_assignname`.